packages/service-registry-api/src/error.rs

```diff
@@ -11,10 +11,16 @@ pub enum ContractError {
 
     #[error(transparent)]
     Overflow(#[from] OverflowError),
-
     #[error(transparent)]
     NonEmpty(#[from] nonempty::Error),
-
+    #[error("verifiers count exceeds u16 limit")]
+    AuthorizedVerifiersExceedu16,
+    #[error("verifiers count is negative")]
+    AuthorizedVerifiersNegative,
```

Comment:
with the changes I proposed the count will be modified immediately where the verifiers state changes, so this becomes even more of an invariant (this can never happen as long as the program runs normally). At that point, I would prefer removing the error and replacing it with an `.expect()`, because if we manage to get to a potentially negative number here something has gone horribly wrong and the call parameters of the contract shouldn't have any influence on it